### PR TITLE
Add query parameters to articles api endpoint

### PR DESCRIPTION
**Pull request by JPaulDev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -63,7 +63,52 @@ describe('GET /api/articles', () => {
...
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number),
+          });
+        });
+      });
+  });
+  it('200: should return an empty array if a topic exists but has no articles associated with it', () => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:91]: great edge case

File: __tests__/app.test.js [hunk 2]
```diff
@@ -75,6 +120,86 @@ describe('GET /api/articles', () => {
...
+      'article_id',
+      'author',
+      'votes',
+      'created_at',
+      'comment_count',
+      'title',
+      'topic',
+    ];
+
+    const promises = sortByValues.map((value) => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js:144]: a nice DRY way to test all your sort by queries!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -75,6 +120,86 @@ describe('GET /api/articles', () => {
...
+  it('400: should return a bad request when an invalid order query is provided', () => {
+    return request(app)
+      .get('/api/articles?order=invalid-order')
+      .expect(400)
+      .then(({ body }) => {
+        const { error } = body;
+
+        expect(error).toHaveProperty('msg', 'Bad Request');
+      });
+  });
```
Comment (roisinoreilly) [inline __tests__/app.test.js:202]: excellent testing 10/10

File: models/article.model.js [hunk 4]
```diff
@@ -1,12 +1,45 @@
...
+    queryString += format("WHERE articles.topic = '%s'", topic);
+  }
+
+  queryString += format(
+    'GROUP BY articles.article_id ORDER BY %I %s;',
+    sort_by,
+    order
+  );
+
+  const { rows: articles } = await db.query(queryString);
```
Comment (roisinoreilly) [inline models/article.model.js:42]: nicely done

**Review by roisinoreilly — APPROVED** [review]
really excellent job with this one, it's probably the most challenging ticket and you smashed it. Your testing is really impressive and all your code is DRY, neat, and nice to read, PR approved!🌟
